src/uct/cuda/cuda_copy/cuda_copy_iface.c

```diff
@@ -204,10 +205,15 @@ static void uct_cuda_copy_event_desc_cleanup(ucs_mpool_t *mp, void *obj)
 {
     uct_cuda_copy_event_desc_t *base = (uct_cuda_copy_event_desc_t *) obj;
     ucs_status_t status;
+    int active;
 
-    status = UCT_CUDA_FUNC(cudaEventDestroy(base->event));
-    if (UCS_OK != status) {
-        ucs_error("cudaEventDestroy Failed");
+    UCT_CUDADRV_CTX_ACTIVE(active);
+
+    if (active) {
+        status = UCT_CUDA_FUNC(cudaEventDestroy(base->event));
+        if (UCS_OK != status) {
```

Comment:
no need for this check and error message - it is already done inside `UCT_CUDA_FUNC`

Reply:
@brminich Actually I didn't change the error handling part of the code. I only added the block around it to check if context is active. In fact, I see that in many places in `cuda_copy` transport, errors are handled outside the function too.

```
$ grep -r UCT_CUDA_FUNC src/uct/cuda/cuda_copy/
src/uct/cuda/cuda_copy/cuda_copy_ep.c:        status = UCT_CUDA_FUNC(cudaStreamCreateWithFlags(&(_strm), cudaStreamNonBlocking)); \
src/uct/cuda/cuda_copy/cuda_copy_ep.c:    status = UCT_CUDA_FUNC(cudaMemcpyAsync(dst, src, length, direction, stream));
src/uct/cuda/cuda_copy/cuda_copy_ep.c:    status = UCT_CUDA_FUNC(cudaEventRecord(cuda_event->event, stream));
src/uct/cuda/cuda_copy/cuda_copy_ep.c:    UCT_CUDA_FUNC(cudaMemcpyAsync((void *)remote_addr, buffer, length,
src/uct/cuda/cuda_copy/cuda_copy_ep.c:    status = UCT_CUDA_FUNC(cudaStreamSynchronize(iface->stream_h2d));
src/uct/cuda/cuda_copy/cuda_copy_ep.c:    UCT_CUDA_FUNC(cudaMemcpyAsync(buffer, (void *)remote_addr, length,
src/uct/cuda/cuda_copy/cuda_copy_ep.c:    status = UCT_CUDA_FUNC(cudaStreamSynchronize(iface->stream_d2h));
src/uct/cuda/cuda_copy/cuda_copy_iface.c:    status = UCT_CUDA_FUNC(cudaEventCreateWithFlags(&(base->event),
src/uct/cuda/cuda_copy/cuda_copy_iface.c:        status = UCT_CUDA_FUNC(cudaEventDestroy(base->event));
src/uct/cuda/cuda_copy/cuda_copy_iface.c:            UCT_CUDA_FUNC(cudaStreamDestroy(self->stream_h2d));
src/uct/cuda/cuda_copy/cuda_copy_iface.c:            UCT_CUDA_FUNC(cudaStreamDestroy(self->stream_d2h));
src/uct/cuda/cuda_copy/cuda_copy_md.c:    status = UCT_CUDA_FUNC(cudaHostRegister(address, length,
src/uct/cuda/cuda_copy/cuda_copy_md.c:    status = UCT_CUDA_FUNC(cudaHostUnregister(address));
```

Reply:
ok, did not notice it is an old code. Anyway this check does not make any sense - it is done to report an error, which is also reported by `UCT_CUDA_FUNC`

Reply:
@brminich I agree that status check is redundant. I've removed for this instance for now. Let me know if I should address others such instances in `uct/cuda/cuda_copy` as part of this PR or a separate one. Thanks

Reply:
thanks, no need for unrelated fixes here